internal/service/odb/cloud_vm_cluster.go

```diff
@@ -576,40 +576,10 @@ func (r *resourceCloudVmCluster) Create(ctx context.Context, req resource.Create
 		return
 	}
 	plan.GiVersion = flex.StringToFramework(ctx, giVersionMajor)
-	odbNetwork = plan.OdbNetworkId
-	if odbNetwork.IsNull() || odbNetwork.IsUnknown() {
-		odbNetwork = plan.OdbNetworkArn
-	}
-	getOdbNetworkInput := odb.GetOdbNetworkInput{
-		OdbNetworkId: odbNetwork.ValueStringPointer(),
-	}
-	odbNetworkOutput, err := conn.GetOdbNetwork(ctx, &getOdbNetworkInput)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			create.ProblemStandardMessage(names.ODB, create.ErrActionWaitingForCreation, ResNameCloudVmCluster, plan.CloudVmClusterId.ValueString(), err),
-			err.Error(),
-		)
-		return
-	}
-	plan.OdbNetworkId = flex.StringToFramework(ctx, odbNetworkOutput.OdbNetwork.OdbNetworkId)
-	plan.OdbNetworkArn = flex.StringToFramework(ctx, odbNetworkOutput.OdbNetwork.OdbNetworkArn)
-	cloudExadataInfra = plan.CloudExadataInfrastructureId
-	if cloudExadataInfra.IsNull() || cloudExadataInfra.IsUnknown() {
-		cloudExadataInfra = plan.CloudExadataInfrastructureArn
-	}
-	getCloudExadataInfraInput := odb.GetCloudExadataInfrastructureInput{
-		CloudExadataInfrastructureId: cloudExadataInfra.ValueStringPointer(),
-	}
-	cloudExadataInfraOutput, err := conn.GetCloudExadataInfrastructure(ctx, &getCloudExadataInfraInput)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			create.ProblemStandardMessage(names.ODB, create.ErrActionWaitingForCreation, ResNameCloudVmCluster, plan.CloudVmClusterId.ValueString(), err),
-			err.Error(),
-		)
-		return
-	}
-	plan.CloudExadataInfrastructureId = flex.StringToFramework(ctx, cloudExadataInfraOutput.CloudExadataInfrastructure.CloudExadataInfrastructureId)
-	plan.CloudExadataInfrastructureArn = flex.StringToFramework(ctx, cloudExadataInfraOutput.CloudExadataInfrastructure.CloudExadataInfrastructureArn)
+	plan.OdbNetworkId = flex.StringToFramework(ctx, createdVmCluster.OdbNetworkId)
+	plan.OdbNetworkArn = flex.StringToFramework(ctx, createdVmCluster.OdbNetworkArn)
+	plan.CloudExadataInfrastructureId = flex.StringToFramework(ctx, createdVmCluster.CloudExadataInfrastructureId)
+	plan.CloudExadataInfrastructureArn = flex.StringToFramework(ctx, createdVmCluster.CloudExadataInfrastructureArn)
```

Comment:
@pooja-choudhury we should be able to remove these lines of code and let the `flex.Flatten(ctx, createdVmCluster, &plan)` handle setting these values. Since the structure names are identical then it should just work

Reply:
Thanks for confirming that the explicit flex function is setting a value that auto flex is not